keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -254,12 +263,138 @@ tried this feature in Alpha, we would have time to fix issues.
 - maxUnavailable greater than 1 with partition and staged pods greater than maxUnavailable
 - maxUnavailable greater than 1 with partition and maxUnavailable greater than replicas
 
+## Test Plan
+For `Alpha`, unit tests and e2e tests will be added to test functionality at both
+with feature flag enabled and disabled. Defaults will be verified so that users
+who donot set this flag are not surprised at all.
+
+
 ## Graduation Criteria
 
 - Alpha: Initial support for maxUnavailable in StatefulSets added. Disabled by default. 
 - Beta:  Enabled by default with default value of 1. 
 
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: MaxUnavailableStatefulSet
+  - Components depending on the feature gate: StatefulSet
+
+###### Does enabling the feature change any default behavior?
+
+No, the default behavior remains the same.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes this feature can be disabled. Once disabled, all existing  StatefulSet will
+revert to the old behavior where rolling update will proceed one pod at a time.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+If the feature is enabled and then later disabled, we will revert the value of 
+maxUnavailable to the default value so that customers are not surprised.
```

Comment:
That's not what we want (unless I misunderstood you) - see the comment above also.

If the field is set in the API, it shouldn't silently disappeared - it's just the controller who should ignore its existence.

Reply:
we start [here](https://github.com/kubernetes/kubernetes/pull/82162/files#diff-a1e25eb50e4b2aed8947c2caa5d6d1ddfef1a5f2f95fdf71f2388c657823ee13R518-R521) and get the value of maxUnavailable from this new field in StatefulSet.

We again [check](https://github.com/kubernetes/kubernetes/pull/82162/files#diff-a1e25eb50e4b2aed8947c2caa5d6d1ddfef1a5f2f95fdf71f2388c657823ee13R528-R529) if the feature flag is disabled we reset maxUnaavailable local variable to 1 and then the rest of the code uses this local variable.

Reply:
The text is misleading, because we won't be reverting the value itself.
Also - it's about disabling - not reenabling.

Can you pleas change to something along the lines with: we will restore the desired behavior for StatefulSets for which the maxunavailable field wasn't deleted after the feature gate was disabled

Reply:
can you actually delete a field after a feature gate was disabled ?